pkg/kubernetes/deploy.js

```diff
@@ -0,0 +1,464 @@
+            $('#namespace_value input[type=text]').val(appdeployer.namespacetxt);
+            deploy_dialog_remove_file_errors();
+            manifest_file.val('');
+            manifest_file.trigger('click');
+
+        });
+
+        $('#namespace_value').on('input', function() { 
+            deploy_dialog_remove_ns_errors();
+            appdeployer.namespacetxt = $('#namespace_value input[type=text]').val();
+            window.setTimeout(check_ns, 2000);
+
+            function check_ns() {
+                if (appdeployer.namespacetxt.trim() === '' || appdeployer.namespacetxt.trim() === 'Custom Namespace' ||
+                    !/^[a-z0-9]+$/i.test(appdeployer.namespacetxt.trim())) {
+                    $('#deploy-app-namespace-field-note').addClass('has-error').show();
+                    valid_ns = false;
+                    display_deploy_button();
+                    return;
+                } else {
+                    valid_ns = true;
+                    display_deploy_button();
+                }
+            }
+        });
+
+        dlg.on('show.bs.modal', function() {
+            appdeployer.namespacetxt = "";
+            valid_ns = false;
+            valid_manifest = false;
+            $('#namespace_value input[type=text]').val('');
+            disable_deploy_button();
+            deploy_dialog_remove_file_errors();
+            deploy_dialog_remove_ns_errors();
+            remove_notifications();
+            //avoid recreating the options
+            if (firstTime) {
+                var template = $('#deploy-app-ns-template').html();
+                Mustache.parse(template);
+                var text = Mustache.render(template, $.extend({
+                    "namespaces": client.namespaces
+                }));
+                ns_selector.html(text);
+                ns_selector.combobox();
+                firstTime = false;
+            }
+            manifest_file_btn.text(_("Select Manifest File...")).addClass('manifest_file_default');
+            ns_selector.combobox();
+            appdeployer.jsondata = "";
+        });
+
+
+        dlg.on('keypress', function(e) {
+            if (e.keyCode === 13)
+                deploy_btn.trigger('click');
+        });
+
+        //focus out also calls change
+        ns_selector.on('change', function() {
+            remove_notifications();
+            deploy_dialog_remove_ns_errors();
+            $('#namespace_value input[type=text]').val(appdeployer.namespacetxt);
+            if(ns_selector.val() === '') {
+                if (appdeployer.namespacetxt.trim() === '' || appdeployer.namespacetxt.trim() === 'Custom Namespace' || 
+                        !/^[a-z0-9]+$/i.test(appdeployer.namespacetxt.trim())) {
+                    $('#deploy-app-namespace-field-note').addClass('has-error').show();
+                    valid_ns = false;
+                    display_deploy_button();
+                    return;
```

Comment:
I see this code being copied about 3 or 4 times here. It should probably be a function.